flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/datagen/DataGeneratorSource.java

```diff
@@ -45,18 +62,34 @@
 	 * @param generator data generator.
 	 */
 	public DataGeneratorSource(DataGenerator<T> generator) {
-		this(generator, Long.MAX_VALUE);
+		this(generator, "generator", Long.MAX_VALUE, null);
 	}
 
 	/**
 	 * Creates a source that emits records by {@link DataGenerator}.
 	 *
 	 * @param generator data generator.
 	 * @param rowsPerSecond Control the emit rate.
+	 * @param numberOfRows Total number of rows to output.
 	 */
-	public DataGeneratorSource(DataGenerator<T> generator, long rowsPerSecond) {
+	public DataGeneratorSource(DataGenerator<T> generator, String name, long rowsPerSecond, Long numberOfRows) {
 		this.generator = generator;
+		this.name = name;
```

Comment:
Do we need to add name?

Reply:
that's a remnant from an old commit, will remove.